discovery/targetgroup/targetgroup_test.go

```diff
@@ -17,75 +17,147 @@ import (
 	"errors"
 	"testing"
 
-	"github.com/prometheus/prometheus/util/testutil"
 	"gopkg.in/yaml.v2"
+	"github.com/prometheus/common/model"
+	
+	"github.com/prometheus/prometheus/util/testutil"
 )
 
 func TestTargetGroupStrictJsonUnmarshal(t *testing.T) {
 	tests := []struct {
-		json          string
-		expectedReply error
+		json            string
+		expectedReply   error
+		expectedTargets []model.LabelSet
```

Comment:
While you're at it, it might be worth extending the test to also cover `Group.Labels`. So instead of checking  `Group.Targets`, check that `Group` instances are equal.

Reply:
Yes, good idea.